### PR TITLE
SYM-7415: Add :SOURCE_SCHEMA and :SOURCE_CATALOG tokens to column router

### PR DESCRIPTION
**Pull request by gp510**
## Summary
- Add `SOURCE_SCHEMA` and `SOURCE_CATALOG` as virtual columns (left-side) and tokens (right-side) in the column match data router
- Enables routing by source schema/catalog from `TriggerHistory` using expressions like `SOURCE_SCHEMA=:EXTERNAL_ID`
- Replaces the fragile MSSQL-specific `log.miner.mssql.populate.external.data.with.schema` workaround (removed in companion symmetric-pro PR)

## Test plan
- [x] 3 new unit tests added: schema equals, schema not-equals, catalog equals
- [x] All 12 `ColumnMatchDataRouterTest` tests pass
- [ ] Manual verification with wildcard schema triggers on MSSQL

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java [hunk 1]
```diff
@@ -122,6 +131,20 @@ public Set<String> routeToNodes(SimpleRouterContext routingContext,
...
+                            nodeIds = runExpression(e, columnValue, sourceSchema, nodes,
+                                    nodeIds, node);
+                        }
+                    } else if (value.equalsIgnoreCase(TokenConstants.SOURCE_CATALOG)) {
+                        String sourceCatalog = dataMetaData.getTriggerHistory() != null
+                                ? dataMetaData.getTriggerHistory().getSourceCatalogName() : null;
+                        for (Node node : nodes) {
+                            nodeIds = runExpression(e, columnValue, sourceCatalog, nodes,
+                                    nodeIds, node);
+                        }
```
Comment (evan-miller-jumpmind) [inline symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java]: Are the checks for `value.equalsIgnoreCase(TokenConstants.SOURCE_SCHEMA)` and `value.equalsIgnoreCase(TokenConstants.SOURCE_CATALOG)` necessary? These would only be necessary if we wanted to support an expression of something like `STATUS=:SOURCE_SCHEMA`, which I'm not sure would make sense.

File: symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java [hunk 2]
```diff
@@ -89,6 +91,13 @@ public Set<String> routeToNodes(SimpleRouterContext routingContext,
                     String column = e.getTokens()[0].trim();
                     String value = e.getTokens()[1];
                     String columnValue = columnValues.get(column);
+                    if (columnValue == null && "SOURCE_SCHEMA".equalsIgnoreCase(column)) {
+                        columnValue = dataMetaData.getTriggerHistory() != null
+                                ? dataMetaData.getTriggerHistory().getSourceSchemaName() : null;
+                    } else if (columnValue == null && "SOURCE_CATALOG".equalsIgnoreCase(column)) {
+                        columnValue = dataMetaData.getTriggerHistory() != null
+                                ? dataMetaData.getTriggerHistory().getSourceCatalogName() : null;
+                    }
```
Comment (evan-miller-jumpmind) [inline symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java:101]: This would prevent users from routing based on the value of a column called `SOURCE_SCHEMA` or `SOURCE_CATALOG`. I know this is an edge case, but maybe it would be better to use the `TokenConstants` here so they're prefixed with a colon.

File: symmetric-core/src/test/java/org/jumpmind/symmetric/route/ColumnMatchDataRouterTest.java
Comment (evan-miller-jumpmind) [inline symmetric-core/src/test/java/org/jumpmind/symmetric/route/ColumnMatchDataRouterTest.java:1]: Can you resolve the Sonar warnings in the test class? I'm not as concerned about the warning in `ColumnMatchDataRouter` since the `routeToNodes()` method probably would have already been detected as a "Brain Method" before your changes.

**Review by evan-miller-jumpmind — DISMISSED** [review]
In addition to my comments, can you add documentation for the new tokens? The documentation is in the `column.ad` file.

File: symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java (hunk 1 above)
Comment (gp510) [reply symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java]: Agreed — removed these branches. The practical use case is always `:SOURCE_SCHEMA` on the left side (e.g. `:SOURCE_SCHEMA=:EXTERNAL_ID`).

File: symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java (hunk 2 above)
Comment (gp510) [reply symmetric-core/src/main/java/org/jumpmind/symmetric/route/ColumnMatchDataRouter.java:101]: Good catch. Changed to use `TokenConstants.SOURCE_SCHEMA` / `TokenConstants.SOURCE_CATALOG` for the comparison, so the expression syntax is now `:SOURCE_SCHEMA=:EXTERNAL_ID` — no collision with actual column names.

File: symmetric-core/src/test/java/org/jumpmind/symmetric/route/ColumnMatchDataRouterTest.java
Comment (gp510) [reply symmetric-core/src/test/java/org/jumpmind/symmetric/route/ColumnMatchDataRouterTest.java:1]: Done — removed `public` modifier from the 3 new test methods to fix the Sonar S5786 warnings.

**Comment by gp510** [comment]
Added documentation for the new tokens in `column.ad` — includes a TIP block listing `:SOURCE_SCHEMA` and `:SOURCE_CATALOG` as virtual columns, plus a full example showing wildcard schema routing with `:SOURCE_SCHEMA=:EXTERNAL_ID`.

File: symmetric-assemble/src/asciidoc/configuration/routers/column.ad
Comment (evan-miller-jumpmind) [inline symmetric-assemble/src/asciidoc/configuration/routers/column.ad]: Here's one more place in the documentation that needs changed. It's no longer true that the first part of the expression is always the column name.

Comment (gp510) [reply symmetric-assemble/src/asciidoc/configuration/routers/column.ad]: Good catch. Updated the intro to say "typically a column name" and mention that `:SOURCE_SCHEMA` and `:SOURCE_CATALOG` virtual column tokens can also be used in place of a column name.

**Review by pavel-jm — APPROVED** [review]
Thank you!

**Review by evan-miller-jumpmind — APPROVED** [review]
